src/renderer-process/js/modulation-matrix-view.js

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 Colin Clark
+
+Licensed under the MIT license.
+https://github.com/colinbdclark/bubbles/raw/master/LICENSE
+*/
+
+"use strict";
+
+fluid.defaults("bubbles.modulationMatrixView", {
+    gradeNames: "fluid.viewComponent",
+
+    layerModulationNames: [
+        "redScale",
+        "blueScale",
+        "greenScale",
+        "opacity",
+        "keyerMin",
+        "keyerMax"
+    ],
+
+    model: {
+        speed: "{videoLayerView}.videoLayer.player.model.rate"
+    },
+
+    dynamicComponents: {
+        modulationView: {
+            createOnEvent: "onModulationViewContainerRendered",
+            type: "bubbles.modulationView",
+            container: "{arguments}.0",
+            options: {
+                modulationName: "{arguments}.1",
+                label: "{arguments}.2"
+            }
+        },
+
+        uniformRelayer: {
+            type: "bubbles.layerUniformRelayer",
+            sources: "{that}.options.layerModulationNames",
+            options: {
+                uniformName: "{source}",
+                layerIdx: "{modulationMatrixView}.options.layerIdx",
+                components: {
+                    relaySource: "{compositor}",
+                    relayTarget: "{modulationMatrixView}"
+                }
+            }
+        }
+    },
+
+    strings: {
+        redScale: "Red",
+        blueScale: "Blue",
+        greenScale: "Green",
+        opacity: "Opacity",
+        keyerMin: "Keyer Min",
+        keyerMax: "Keyer Max",
+        speed: "Speed"
+    },
+
+    markup: {
+        modulationViewContainer: "<div class='bubbles-modulator'></div>"
+    },
+
+    events: {
+        onModulationViewContainerRendered: null
+    },
+
+    listeners: {
+        "onCreate.renderContainers": {
+            funcName: "bubbles.modulationMatrixView.renderModViewContainers",
+            args: ["{that}"]
+        }
+    }
+});
+
+bubbles.modulationMatrixView.renderModViewContainers = function (that) {
+    fluid.each(that.options.strings, function (label, modulationName) {
+        bubbles.modulationMatrixView.renderModViewContainer(that,
+            modulationName, label);
+    });
+};
+
+bubbles.modulationMatrixView.renderModViewContainer = function (that, modulationName, label) {
+    var modViewContainer = $(that.options.markup.modulationViewContainer);
```

Comment:
You may be interested to shift responsibility for markup and rendering into the modulationView component itself by means of the (somewhat undocumented) support for "fluid.containerRenderingView" in the new "protoframework" - https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/NewViewSupport.js#L57 - you can see a usage at https://github.com/fluid-project/infusion/blob/master/src/components/orator/js/Orator.js#L127

Reply:
Oooh, this is great—all of this "child container rendering" logic scattered around in my parent components has been really bothering me. Can this grade be used with mainline Infusion dev releases, or will I need to move to the potentia branch in order to use it?

Reply:
The stuff linked is already in trunk, it's rickety (e.g. doesn't seem to have support for taking the markup back out again on "destroy" but then neither did you : P) but basically usable today

Reply:
That's so great, thanks for the heads up. It's nice to have a few supports for us self-rendering dinosaurs. :)